packages/modules-sdk/src/loaders/__tests__/medusa-module.ts

```diff
@@ -0,0 +1,61 @@
+import {
+  InternalModuleDeclaration,
+  MODULE_RESOURCE_TYPE,
+  MODULE_SCOPE,
+} from "@medusajs/types"
+import { MedusaModule } from "../../medusa-module"
+
+const mockRegisterMedusaModule = jest
+  .fn()
+  .mockImplementation(() => Promise.resolve([]))
+const mockModuleLoader = jest.fn().mockImplementation(() => Promise.resolve({}))
+
+jest.mock("./../../loaders", () => ({
+  registerMedusaModule: jest
+    .fn()
+    .mockImplementation((...args) => mockRegisterMedusaModule()),
+  moduleLoader: jest.fn().mockImplementation((...args) => mockModuleLoader()),
+}))
+
+describe("Medusa Module", () => {
+  beforeEach(() => {
+    jest.resetModules()
+    jest.clearAllMocks()
+  })
+
+  it.only("MedusaModule bootstrap - Singleton instances", async () => {
+    await MedusaModule.bootstrap(
+      "moduleKey",
+      "@path",
+      {
+        scope: MODULE_SCOPE.INTERNAL,
+        resources: MODULE_RESOURCE_TYPE.ISOLATED,
+        resolve: "@path",
+        options: {
+          abc: 123,
+        },
+      } as InternalModuleDeclaration,
+      {}
+    )
+
+    expect(mockRegisterMedusaModule).toBeCalledTimes(1)
+    expect(mockModuleLoader).toBeCalledTimes(1)
```

Comment:
question: Are these required? Shouldn't this test run the bootstrapping twice and make the assertion once?

Reply:
The test below is doing it, calling again the bootstrap and the calledTimes should continue to be 1.

Reply:
Yes, I understand. I would just think asserting only once, in the end, yields the same result. But no need to go back and forth about this.